march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -230,6 +231,12 @@ void MarchHardwareInterface::write(const ros::Time& /* time */, const ros::Durat
   {
     // Enlarge joint_effort_command because ROS control limits the pid values to a certain maximum
     joint_effort_command_[i] = joint_effort_command_[i] * 1000.0;
+    if (abs(joint_effort_command_copy_[i] - joint_effort_command_[i]) > MAX_EFFORT_CHANGE)
+    {
+      joint_effort_command_copy_[i] =
+          joint_effort_command_copy_[i] +
+          copysign(MAX_EFFORT_CHANGE, joint_effort_command_[i] - joint_effort_command_copy_[i]);
```

Comment:
You write to the `joint_effort_command_copy_` here, however, you never read this value again. Instead you write over it with the `joint_effort_command` on line 254. I think you should write to `joint_effort_command_` here.